### PR TITLE
feat(appsync): add Construct for AppSync HTTP DataSource

### LINKED ISSUE #8007: feat(appsync): add Construct for AppSync HTTP DataSource
I implemented a `Construct` level of `HTTP DataSource` for `appsync`.

image

### Use Case

Currently `aws-appsync` module provides these:

- [x] None
- [x] DynamoDB
- [x] Lambda

But not these:

- [ ] **HTTP**
- [ ] Elastisearch
- [ ] RDS

### Proposed Solution

HTTP data source only takes http endpoint. (just extends `appsync.BaseDataSource`)

image

```ts
/**
 * An AppSync datasource backed by a http endpoint
 */
export class HttpDataSource extends appsync.BaseDataSource {
  constructor(scope: cdk.Construct, id: string, props: HttpDataSourceProps) {
    super(scope, id, props, {
      httpConfig: {
        endpoint: props.endpoint,
      },
      type: 'HTTP',
    })
  }
}
```

### Other

And I also added some resolver mapping template examples for HTTP data source:

image
image

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by civilizeddev**
### Commit Message
feat(appsync): add Construct for AppSync HTTP DataSource

* ADD GraphQLApi.addHttpDataSource()
* ADD interface HttpDataSourceProps
* ADD class HttpDataSource

Closes #8007 
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by MrArnoldPalmer** [comment]
The http data source is super useful and I think we should merge it so users can start playing with it. This PR has also shown the need for more tools to allow easier building and composition of mapping templates. Some of the mapping templates added here probably aren't going to be widely useful. What about users that want to forward some headers, and some of the request body?

I'm hesitant to include these Mapping templates for this reason. If users start to rely on them and ultimately we decide not to support them that isn't going to be a great experience. Wrote this issue to work towards a more long term solution https://github.com/aws/aws-cdk/issues/8216. For now I think we should limit the mapping templates we include to a GET/POST/PATCH/DELETE methods forwarding all headers and body (as query params for GET/DELETE). This obviously makes a lot of assumptions about the user's data source but writing tons of various static methods on MappingTemplate is starting to get unwieldy and users can resort to `.fromString` if they have more specific needs.

File: packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts [hunk 1]
```diff
@@ -1027,6 +1066,128 @@ export abstract class MappingTemplate {
     return this.fromString('$util.toJson($ctx.result)');
   }
 
+  /**
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts:1069]: Lets limit the request mapping templates to `httpGet`, `httpPost`, `httpPatch`, and `httpDelete`. We can take additional arguments to allow the user to specify whether they want to pass headers/body.

File: packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts [hunk 2]
```diff
@@ -1027,6 +1066,128 @@ export abstract class MappingTemplate {
...
+          ## If response is not 200, append the response to error block.
+          $utils.appendError($ctx.result.body, "$ctx.result.statusCode")
+      #end
+    `);
+  }
+
+  /**
+   * HTTP Response Mapping Template Sample: Return XML reponse as json
+   */
+  public static httpReturnXmlResponseAsJson(): MappingTemplate {
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts:1174]: I think we should take out this method for now as well. I may be mistaken in thinking this wont be widely used, but its easier to add later if users ask for it then it is to remove if we decide we want to.

File: packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts (hunk 1 above)
Comment (civilizeddev) [reply packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts]: I think giving an example in README is enough for now. I removed them.

**Pull request merged**